Add logformat journald to localfile reference (#7202)
wazuh/wazuh-documentation: source/user-manual/reference/ossec-conf/localfile.rst

  The `eventchannel` format already provides a way to filter logs through queries. Therefore, `ignore` and `restrict` settings don't apply to this format.

filter
^^^^^^

The `filter` tag is used to include PCRE2 regex filters for selectively collecting logs based on specific fields within `journald`.
Each filter must specify a field and a regex pattern. The `ignore_if_missing` attribute can be used to indicate whether to ignore logs where the specified field is missing.

+--------------------+---------------------------------------------------------------+
| **Default Value**  | n/a                                                           |
+--------------------+---------------------------------------------------------------+
| **Allowed values** | Any `PCRE2 <regex.html#pcre2-syntax>`_ expression.            |
+--------------------+---------------------------------------------------------------+

Use the `field` attribute to define in which journald field to apply the regex (Mandatory).

+-----------------------+--------------------------------------------------------------------------------------------------------------+

> | **ignore_if_missing** | When the attribute `ignore_if_missing` is set to `yes` it ignores the filter if the field does not exist.    |

Author: I take your suggestion and add the description to not.

+                       +------------------+-------------------------------------------------------------------------------------------+
|                       | Default value    | no                                                                                        |
|                       +------------------+-------------------------------------------------------------------------------------------+
|                       | Allowed values   | no, yes                                                                                   |
+-----------------------+------------------+-------------------------------------------------------------------------------------------+

> Configuration example:

Author: Added in the last commit with a short description

.. code-block:: xml

    <!-- For monitoring log files -->
    <localfile>
      <location>journald</location>
      <log_format>journald</log_format>
      <filter field="_SYSTEMD_UNIT">^cron.service$</filter>
      <filter field="PRIORITY" ignore_if_missing="yes">[0-3]</filter>
    <localfile>

.. note::

>     Filters within the same `<localfile>` block follow an AND logic, while multiple blocks are evaluated in OR logic regarding log collection.

Author: ditto

> Configuration examples

Author: I'm not sure about this, it would involve removing the syslog blocks with the configuration as these files are no longer collected as localfile but are collected by journald.

  The `eventchannel` format already provides a way to filter logs through queries. Therefore, `ignore` and `restrict` settings don't apply to this format.

filter
^^^^^^

Collects ``journald`` logs selectively by filtering specific fields.

You must specify a PCRE2 regex pattern as your filter. Use the ``field`` attribute to define the journald field where to apply the regular expression.

+--------------------+---------------------------------------------------------------+
| **Default Value**  | n/a                                                           |
+--------------------+---------------------------------------------------------------+
| **Allowed values** | Any :ref:`PCRE2 <pcre2_syntax>` expression.                   |
+--------------------+---------------------------------------------------------------+

You can use the ``ignore_if_missing`` attribute to ignore logs without the specified field.

+-----------------------+--------------------------------------------------------------------------------------------------------------+

> | **ignore_if_missing** | When the attribute `ignore_if_missing` is set to `yes` it ignores the filter if the field does not exist.    |

Reviewer: ```suggestion
| **ignore_if_missing** | When set to ``yes``, it accepts logs without the specified field. Conversely, when set to ``no``,            |
```

> |                       | If is set to `no` and the field does not exist, the filter will fail and the log will be ignored.            |

Reviewer: ```suggestion
|                       | logs without the specified field fail to meet the filtering criteria and are disregarded.                    |
```

+                       +------------------+-------------------------------------------------------------------------------------------+
|                       | Default value    | no                                                                                        |
|                       +------------------+-------------------------------------------------------------------------------------------+
|                       | Allowed values   | no, yes                                                                                   |
+-----------------------+------------------+-------------------------------------------------------------------------------------------+

> 

Reviewer: ```suggestion
```

> In the following example configuration the journald logs will be collected if any of the following conditions are met:

Reviewer: ```suggestion
In the following configuration example Wazuh collects the ``journald`` logs if any of the following conditions are met.

```

>   - Field `_SYSTEMD_UNIT` exists and its value is `ssh.service`.

Reviewer: ```suggestion
-  The field ``_SYSTEMD_UNIT`` is present with the value ``ssh.service``.
```

>   - Field `_SYSTEMD_UNIT` exists and its value is `cron.service` and if the field `PRIORITY` exists, it has the value 0,1,2 or 3.

Reviewer: ```suggestion
-  The field ``_SYSTEMD_UNIT`` is present with the value ``cron.service`` and the field ``PRIORITY`` is present with the value ``0``, ``1``, ``2``, or ``3``.
```

.. code-block:: xml

    <!-- For monitoring log files -->
    <localfile>
      <location>journald</location>
      <log_format>journald</log_format>
      <filter field="_SYSTEMD_UNIT">^ssh.service$</filter>
    <localfile>

    <localfile>
      <location>journald</location>
      <log_format>journald</log_format>
      <filter field="_SYSTEMD_UNIT">^cron.service$</filter>
      <filter field="PRIORITY" ignore_if_missing="yes">[0-3]</filter>
    <localfile>

.. note::

>     Filters within the same `<localfile>` block follow an AND logic, while multiple blocks are evaluated in OR logic regarding log collection.

Reviewer: ```suggestion

   Filters within the same ``<localfile>`` block follow an AND logic, while multiple blocks are evaluated in OR logic regarding log collection.
```
